cs4660/graph/graph.py

```diff
@@ -95,22 +105,54 @@ def __init__(self):
         self.adjacency_list = {}
 
     def adjacent(self, node_1, node_2):
-        pass
+        for edge in self.adjacency_list[node_1]:
+            if edge.to_node == node_2:
+                return True
+        return False
 
     def neighbors(self, node):
-        pass
+        neighbors_list = []
+        if node in self.adjacency_list:
+            for x in self.adjacency_list[node]:
+                neighbors_list.append(x.to_node)
+        return neighbors_list
 
     def add_node(self, node):
-        pass
+        if node not in self.adjacency_list:
+            self.adjacency_list[node] = []
+            return True
+        else:
+            return False
 
     def remove_node(self, node):
-        pass
+        if node in self.adjacency_list:
+            for x in self.adjacency_list.keys():
+                key_value = self.adjacency_list[x]
+                for y in key_value:
+                    if y.to_node == node:
+                        self.adjacency_list[x].remove(y)
+            self.adjacency_list.pop(node)
+            return True
+        else:
+            return False
 
     def add_edge(self, edge):
-        pass
+        for x in self.adjacency_list[edge.from_node]:
+            if x == edge:
+                return False
+        if self.adjacency_list[edge.from_node] == {}:
```

Comment:
unnecessary check as you never assign the value of dictionary to be empty dictionary `{}`.